doc/manual/src/contributing/documentation.md

```diff
@@ -206,3 +206,22 @@ or inside `nix-shell` or `nix develop`:
 # make internal-api-html
 # xdg-open ./outputs/doc/share/doc/nix/internal-api/html/index.html
 ```
+
+## C API documentation (experimental)
+
+[C API documentation] is available online.
+You can also build and view it yourself:
```

Comment:
Would be good to also have this info in a more visible place, but that's not required for this PR necessarily.